### DISCUSSION TITLE
some questions about param num_workers

### DISCUSSION THREAD

**Discussion by Pomelochen** [alibaba/GraphScope#2844] [Q&A]
分别在hosts模式和k8s模式下对参数num_workers进行了一些图分析算法的测试，例如pagerank，sssp，clustrring等，有一些疑问。

> 在hosts模式下，后台有num_workers指定数量的grape_engine进程，并且cpu都运行在100%，但是计算速度最快的是num_workers=1，并且参数越大，计算速度越慢。

> 在k8s模式下，后台也拉起了指定数量的pods，但同样的问题，num_workers的大小对于计算速度几乎毫无帮助。
这个参数的实际意义是什么？还是我的使用方式不对？

**Comment by siyuan0322** [comment]
你好，一般来讲，图会被分为 num_workers 个分区，在计算过程中多分区会涉及到通信的过程。 如果单 worker 可以载完全图，那么效果会比再拆分多分区要好。 多worker 的情况一般为单台机器内存不足够载入全图的情况，比如数百亿节点的图。
